### PR TITLE
chore: deprecate job move within priority group

### PR DESCRIPTION
**Pull request by kkunapuli**
## Ticket

[RM-360]

## Description

Deprecate job move within the same priority group. Jobs can still be re-ordered via CLI command by updating their priority. 

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[RM-360]: https://hpe-aiatscale.atlassian.net/browse/RM-360?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/release-notes/deprecate-job-move.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+:orphan:
+
+**Deprecations**
+
+-  Agent and Kubernetes Resource Manager: Jobs can no longer be moved within the same priority
+   group. Update a job's priority via CLI command to move a job ahead, or behind, of another job.
+   Visit :ref:`modify-job-queue-cli` for details. Deprecation was announced in 0.33.0.
```
Comment (tara-det-ai) [inline docs/release-notes/deprecate-job-move.rst:7]: - Agent and Kubernetes Resource Manager: Jobs can no longer be moved within the same priority group. To reposition a job, update its priority using the CLI. For detailed instructions, visit :ref:`modify-job-queue-cli`. This change was announced in version 0.33.0.

**Review by tara-det-ai — APPROVED** [review]
added suggestion

File: master/internal/job/jobservice/jobservice.go [hunk 2]
```diff
@@ -206,19 +206,9 @@ func (s *Service) applyUpdate(update *jobv1.QueueControl) error {
 		}
 		return j.SetResourcePool(action.ResourcePool)
 	case *jobv1.QueueControl_AheadOf:
-		return s.rm.MoveJob(sproto.MoveJob{
-			ID:           jobID,
-			Anchor:       model.JobID(action.AheadOf),
-			Ahead:        true,
-			ResourcePool: j.ResourcePool(),
-		})
+		return fmt.Errorf("action not supported - update priority to move job")
```
Comment (NicholasBlaskey) [inline master/internal/job/jobservice/jobservice.go:209]: we should update the proto definition to no longer have these options

**Review by NicholasBlaskey — APPROVED** [review]

File: master/internal/job/jobservice/jobservice.go (hunk 2 above)
Comment (kkunapuli) [reply master/internal/job/jobservice/jobservice.go:209]: If we leave it like this, we can have a more informative error message though right?

I was thinking we could leave it until the UI changes are done to actually remove the option.

Comment (kkunapuli) [reply master/internal/job/jobservice/jobservice.go:209]: The UI updates need to be prioritized. That could be in a future release. My understanding from product is that it was an acceptable trade-off.

https://hpe-aiatscale.slack.com/archives/C06GMG83ZE0/p1718299032071299?thread_ts=1718298903.798219&cid=C06GMG83ZE0

I agree with marking the proto definition as deprecated.

Comment (NicholasBlaskey) [reply master/internal/job/jobservice/jobservice.go:209]: Shouldn't the UI happen removal happen first? 

Can we just wait till the UI removal happens then?

Comment (kkunapuli) [reply master/internal/job/jobservice/jobservice.go:209]: I don't think I understand what you're suggesting. Are you suggesting we shouldn't merge this PR until the corresponding front-end work is completed?

My understanding from previous conversations is that we were going to remove as much as possible from the backend, and leave it as a backlog task to update the front-end. https://hpe-aiatscale.slack.com/archives/C06GMG83ZE0/p1718311967982659?thread_ts=1718298903.798219&cid=C06GMG83ZE0

Comment (NicholasBlaskey) [reply master/internal/job/jobservice/jobservice.go:209]: That previous conversation is talking about a slightly different situation. The previous conversation assumed that Kubernetes would still support this feature and the backlog ticket is to grey out it for agentrm / slurm cases. This change now has removed the feature from all resource managers. I think there is a gap between having a button that only works in certain situations verse a button that never works.

Have we asked the frontend team if by the release would be a reasonable deadline for their work on this?

File: docs/release-notes/deprecate-job-move.rst (hunk 1 above)
Comment (kkunapuli) [reply docs/release-notes/deprecate-job-move.rst:7]: note to self: adjust priority in CLI or webui

**Pull request merged**